### ISSUE TITLE
`unnest` won't accept a `ReadonlyArray` as argument

### ISSUE THREAD

**Issue by abaco** [TylorS167/167#15]
**Code to reproduce the issue:**

```typescript
const nested: ReadonlyArray<ReadonlyArray<number>> = [[1]];
const flat = unnest(nested);
```

**Expected behavior:**

The code compiles.

**Actual behavior:**

Compilation fails at `unnest` with:
```
Argument of type 'ReadonlyArray<ReadonlyArray<number>>' is not assignable
to parameter of type '(ReadonlyArray<number> | ReadonlyArray<number>[])[]'.
```

As a workaround I'm using this:

```typescript
const flat = chain(id, nested);
```

**Versions of packages used:**

`0.42.0`

**Comment by TylorS** [comment]
Hello @abaco, thanks for reporting this bug. In the meantime you may be able to use `flatten` which like `unnest` except works at unlimited depth where `unnest` works at depth of 1.

**Comment by TylorS** [comment]
This has been fixed in v0.43.0
